### PR TITLE
feat: circles support to groupfolders

### PR DESCRIPTION
**Pull request by ArtificialOwl**
During the configuration process, the front-end can send a circleId instead of a groupId, using the exact same request on the back-end.
The back-end will decide if the id is about a Circle or a Nextcloud Group, by checking that a circle with the id exists or not. If circle does not exists, then the id is the id of a group.

~~The only changes is related to the application returns when listing the groupfolders.
Before this patch, the entry `groups` entry was an array with key=groupId, value=permissions.
Now the value is an array that contains displayName and permissions:~~

Edit: moved new output to a new key to maintain compatibility
```
{
    "id": 1,
    "mount_point": "test_gf001",
    "group": { "admin": 1, "u3HLVORxw8E5IuCZ2UYudIvwDjGIpDa": 7},
    "group_details": {
        "admin": {
            "displayName": "admin",
            "permissions": 1,
            "type": "group"
        },
        "u3HLVORxw8E5IuCZ2UYudIvwDjGIpDa": {
            "displayName": "sub circle",
            "permissions": 7,
            "type": "circle"
        }
    },
    "quota": -3,
    "size": 0,
    "acl": false,
    "manage": []
}
```

When stored in the database, `circle_id` is used for circle, `group_id` for group.

---

- few php8 constructors
- add an endpoint to get available circles to the current viewer. [1]
- add an event listener on circle deletion; to remove any application to all groupfolders based on the singleId of the removed circle.
- change on the applicableMap
- improve the `occ` commands `:group` and `:list` to fit the edit
- update some SQL request to search folders based on the session users' memberships (to circles)
- migration: add `circle_id` to `group_folders_group` and update the unique index

[1] Available Circles:
  - Circles marked as publicly visible.
  - Circles the current viewer is a direct or inherited member of.

Fix https://github.com/nextcloud/groupfolders/issues/2351
Fix https://github.com/nextcloud/groupfolders/issues/2350
Fix https://github.com/nextcloud/groupfolders/issues/2349

### CODE REVIEW

**Comment by icewind1991** [comment]
changed the api controller to maintain backwards compatibility by moving the new group info to a new field and putting the old field in the old format

**Comment by icewind1991** [comment]
code looks ok, basic functionality works (you have to add circles from the cli atm, but you can manager permissions from the webui)

**Comment by ArtificialOwl** [comment]
> changed the api controller to maintain backwards compatibility by moving the new group info to a new field and putting the old field in the old format

This will only affect the controller API, the result from occ command (i.e `groupfolders:list --output=json`) wont keep backwards compatibility. 
Are we okay with that ?

File: tests/psalm-baseline.xml [hunk 1]
```diff
@@ -17,14 +17,42 @@
...
+      <code>$queryHelper = $circlesManager->getQueryHelper();</code>
+      <code>$queryHelper = $circlesManager->getQueryHelper();</code>
+      <code>$queryHelper = $circlesManager->getQueryHelper();</code>
+      <code>$circlesManager->startSuperSession();</code>
+      <code>@param CirclesQueryHelper|null $queryHelper</code>
+      <code>$circlesManager->startSuperSession();</code>
+      <code>$circlesManager->getCircle($groupId, $probe);</code>
+      <code>$circlesManager->stopSession();</code>
+      <code>@param CirclesManager|null $circlesManager</code>
+    </UndefinedClass>
```
Comment (skjnldsv) [inline tests/psalm-baseline.xml:45]: We should fix this instead of committing the new issues :)

**Review by skjnldsv — CHANGES_REQUESTED** [review]

**Comment by icewind1991** [comment]
> > changed the api controller to maintain backwards compatibility by moving the new group info to a new field and putting the old field in the old format
> 
> This will only affect the controller API, the result from occ command (i.e `groupfolders:list --output=json`) wont keep backwards compatibility. Are we okay with that ?

also changed that, good catch

File: tests/psalm-baseline.xml (hunk 1 above)
Comment (icewind1991) [reply tests/psalm-baseline.xml:45]: This is (mostly?) undefined bits from the circles app, so we would need to add stubs for it

**Review by icewind1991 — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**